itk_dev_shared_components/kmd_nova/authentication.py

```diff
@@ -0,0 +1,51 @@
+"""This module contains functionality to authenticate against the KMD Nova api."""
+
+from datetime import datetime, timedelta
+
+import requests
+
+
+# pylint: disable-next=too-few-public-methods
+class NovaAccess:
+    """An object that handles access to the KMD Nova api."""
+    def __init__(self, client_id: str, client_secret: str, domain: str = "https://cap-novaapi.kmd.dk") -> None:
+        self.client_id = client_id
+        self.client_secret = client_secret
+        self._bearer_token, self.token_expiry_date = self._get_new_token()
+        self.domain = domain
+
+    def _get_new_token(self) -> tuple[str, datetime]:
+        """
+        This method requests a new token from the API.
+        When the token expiry date is passed, requests with the token to the API will return HTTP 401 status code.
+
+        Returns:
+            tuple: token and expiry datetime
+
+        Raises:
+            requests.exceptions.HTTPError: If the request failed.
+        """
+
+        url = "https://novaauth.kmd.dk/realms/NovaIntegration/protocol/openid-connect/token"
+        payload = f"client_secret={self.client_secret}&grant_type=client_credentials&client_id={self.client_id}&scope=client"
+        headers = {'Content-Type': 'application/x-www-form-urlencoded'}
+
+        response = requests.post(url, headers=headers, data=payload, timeout=60)
+        response.raise_for_status()
+        bearer_token = response.json()['access_token']
+        token_life_seconds = response.json()['expires_in']
```

Comment:
Consider introducing a variable to prevent calling `response.json` twice.

Or maybe rewrite to something like

```suggestion
        bearer_token, token_life_seconds = itemgetter('access_token', 'expires_in')(response.json())
```

I haven't tested this myself, but stolen it from https://stackoverflow.com/a/52083390/2502647 all by myself. It looks cool!